keel-lemur/src/main/kotlin/com/netflix/spinnaker/keel/lemur/LemurCertificateResolver.kt

```diff
@@ -21,20 +22,28 @@ class LemurCertificateResolver(
         listeners = resource.spec.listeners.mapTo(mutableSetOf()) { listener ->
           listener.certificate?.let { name ->
             listener.copy(
-              certificate = findCurrentCertificate(name)
+              certificate = runBlocking { findCurrentCertificate(name) }
             )
           } ?: listener
         }
       )
     )
 
-  private fun findCurrentCertificate(name: String) =
-    runBlocking {
-      val certificate = lemurService.certificateByName(name).items.first()
-      if (certificate.active) {
-        certificate.name
-      } else {
-        certificate.replacedBy.first { it.active }.name
-      }
+  private suspend fun findCurrentCertificate(name: String): String {
+    val certificate = lemurService.certificateByName(name).items.first()
```

Comment:
`lemurService.certificateByName(name).items` could also be empty here and so `.first()` would throw `NoSuchElementException`, right? I know you mentioned you checked the current certs, but I think it wouldn't hurt to add a check here.

Reply:
Yeah, you're right. Fixed